src/main/java/www/purple/mixxy/controllers/LoginLogoutController.java

```diff
@@ -61,19 +66,47 @@ public Result logout(Context context) {
 	///////////////////////////////////////////////////////////////////////////
 	// Login
 	///////////////////////////////////////////////////////////////////////////
-    public Result login() {
-    	GoogleAuthHelper helper = new GoogleAuthHelper();
-    	return Results.redirect(helper.buildLoginUrl());
+    public Result login(@Param("provider") String provider) {
+    	
+    	if(provider.equals(OAuthProviders.GOOGLE)) {
```

Comment:
This can be a `switch`/`case` here (remember, we'll be adding DeviantArt, and maybe other providers if doing so is easy).